solr/core/src/java/org/apache/solr/update/DirectUpdateHandler2.java

```diff
@@ -972,27 +964,22 @@ private void updateDocOrDocValues(AddUpdateCommand cmd, IndexWriter writer, Term
       log.debug("updateDocValues({})", cmd);
       writer.updateDocValues(updateTerm, fieldsToUpdate.toArray(new Field[fieldsToUpdate.size()]));
     } else {
-      updateDocument(cmd, writer, updateTerm);
+      updateDocument(cmd, docs, writer, updateTerm, isBlock);
```

Comment:
IMO of updateDocument is a method then the code above underneath cmd.isInPlaceUpdate should be a method as well -- `updateDocValues` to be balanced.  Or inline updateDocument.